[cpp] Sanitize identifier names

Jauler opened the pull request:
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.1.x`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Proposed fix for #621 move identifier sanitizing to AbstractCppCodegen.java to provide those functions for other cpp codegens if needed. I wanted to avoid removing other cpp generators overloads as I have no setup for testing those.

@ravinikam (2017/07) @stkrwork (2017/07) @fvarose (2017/11) @etherealjoy (2018/02) @martindelille (2018/03)

wing328 commented:
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

etherealjoy commented:
@Jauler 
Do you wish to clean up duplicate code in the sub classes?

Jauler commented:
Hey,
While I was cleaning up duplicated code, I had some observations:

- CppTizen had already implemented some sanitizing, although It simply deleted invalid characters and sanitizeNames() replaces them with "_". I changed CppTizen to use default behaviour.
- CppQt on toApiName and toModelName methods prepended modelNamePrefix to the name. I believe this is good fit for default behaviour, therefore I added that to AbstractCppCodegen.
- CppQt in toParamName and toVarName converts to snake_case, does not allow all caps etc.  I was not sure why it is needed, so I left this behaviour untouched.

wing328 commented:
@Jauler thanks for cleaning up the duplicated code. Please also run the shell script (./bin/) or Windows batch files (.\bin\windows) to update the Petstore samples for the affected generators (e.g. Tizen) to see if there're any changes to the samples as a result of the cleanup.

Jauler commented:
Good Catch!
I missed the fact, that tizen codegen is not extending AbstractCppCodegen class which caused some unintended differences in code generation.

I believe that making tizen extend AbstractCppCodegen is better suited for separate PR and adding that I have no tizen setup, I would like to be on the safe side, therefore I simply reverted changes done to tizen.

wing328 commented:
@Jauler PR merged into master. Thanks for your contribution.

I've created https://github.com/OpenAPITools/openapi-generator/issues/674 to track extending TizenCodegen with AbstractCppCodegen

etherealjoy commented:
I will also remove the override in the Qt5 Server subclass which is extending the `AbstractCppCodegen`.
